containers/init/ready/ready.go

```diff
@@ -125,83 +118,85 @@ func findDriverPort(pod *corev1.Pod) int32 {
 	return DefaultDriverPort
 }
 
-// WaitForReadyPods blocks until pods with matching label selectors are ready.
+// WaitForReadyPods blocks until all worker pods belonging to the load test are
+// ready.
 // It accepts a context, allowing a timeout or deadline to be specified. When
 // all pods are ready, it returns a slice of strings with the IP address and
-// driver port for each matching pod. The order will match the label selectors.
-//
-// The syntax for the selectors is defined in the Parse function documented at
-// pkg.go.dev/k8s.io/apimachinery/pkg/labels.
+// driver port for each matching pod. server pod would come before client pod.
 //
 // The driver port is determined by searching the pod for a container with a TCP
 // port named "driver". If there is no port named "driver" exposed on any of the
 // matching pod's containers, the value of DefaultDriverPort will be used.
 //
 // If the timeout is exceeded or there is a problem communicating with the
 // Kubernetes API, an error is returned.
-func WaitForReadyPods(ctx context.Context, pl PodLister, sels []string) ([]string, error) {
+func WaitForReadyPods(ctx context.Context, lg LoadtestGetter, pl PodLister, testName string) ([]string, error) {
 	timeoutsEnabled := true
 	deadline, ok := ctx.Deadline()
 	if !ok {
 		timeoutsEnabled = false
 		log.Printf("no timeout is set; this could block forever")
 	}
-
-	selectors, err := parseSelectors(sels)
-	if err != nil {
-		return nil, err
-	}
-
-	var podAddresses []string
-	for range selectors {
-		podAddresses = append(podAddresses, "")
-	}
-
-	matchCount := 0
+	var loadtest *grpcv1.LoadTest
+	var clientPodAddresses []string
+	var serverPodAddresses []string
+	clientMatchCount := 0
+	serverMatchCount := 0
```

Comment:
nit: We should place a space between lines 139 and 140 since they serve as two logically distinct purposes. Or, we could move lines 140-144 to be before line 134. Placing `var` declarations at the top is usually a good thing. If the function is super long, it sometimes helps to place them closer to where they are used though. The only other time that they generally don't belong as the very first thing is when there is a conditional that would prevent them from ever being needed. In this situation, the conditional in lines 136-139 does not return... so there is no reason they cannot be at the top for clarity.

Reply:
Done, appreciate it!

Reply:
Placing variables close to where they are used is generally a good thing too, so I wouldn't say that there is a preference for placing them at the top.